zebra-chain/src/sapling/spend.rs

```diff
@@ -85,3 +85,16 @@ impl ZcashDeserialize for Spend {
         })
     }
 }
+
+impl std::cmp::PartialEq for Spend<PerSpendAnchor> {
```

Comment:
Can you Derive Eq and PartialEq instead of implementing them manually?

Reply:
If we can't derive, how can we make sure we remember to add any extra fields to this equality check?

Also if we're not deriving, let's add this comment.
```rust
    // `Spend`s with different anchor variants can never be equal.
```